web/src/engine/FetchProvider.ts

```diff
@@ -55,8 +55,8 @@ export function FetchCSS<T extends HTMLElement>(request: FetchRequest, query: st
     return fetchCSS<T>(request, query);
 }
 
-let fetchGraphQL: <TResult>(request: FetchRequest, operationName: string, query: string, variables : string) => Promise<TResult> = fail;
-export function FetchGraphQL<TResult>(request: FetchRequest, operationName: string, query: string, variables : string): Promise<TResult> {
+let fetchGraphQL: <TResult>(request: FetchRequest, operationName: string, query: string, variables: unknown) => Promise<TResult> = fail;
+export function FetchGraphQL<TResult>(request: FetchRequest, operationName: string, query: string, variables: unknown): Promise<TResult> {
```

Comment:
Type `unknown` is quite an anti-pattern for using with type definitions.
Consider using an appropriate type or add JSDoc comment for the method parameter.

Reply:
The thing is, by essence, the "variables" object is just an object and its definition depends on the request.

Reply:
I don't think its just an object, it must be JSON serialisable e.g., like the [JSONElement](https://github.com/manga-download/haruneko/blob/fbec2edcda3c2f1bd02f8a3cc7064a8a307b947f/web/src/engine/ipc/InterProcessCommunication.ts#L1) type definition

Reply:
Its basically { x : boolean | Array | string | number } i guess, with any number of property. It could be empty too. I am wondering which type would be ok, or how to define one accuratly. Maybe JSONElement would fit ?